javascript/organizing_your_javascript_code/factory_functions_and_module_pattern.md

```diff
@@ -1,305 +1,250 @@
 
-const jeff = personFactory('jeff', 27);
+In ECMAScript 6, the keywords `let` and `const` were introduced. While `var` variables were function scoped, these allow you to define variables that are **block scoped** - basically, scoping the variable to only be available within the closest set of `{ curly braces }` in which it was defined. These braces can be those of a `for` loop, `if-else` condition, or any other similar construct, and are called, a block. Let's see an example to sum this all up.
 
-console.log(jeff.name); // 'jeff'
+~~~javascript
+let globalAge = 23; // This is a global variable
 
-jeff.sayHello(); // calls the function and logs 'hello!'
-~~~
+function printAge (age) { // This is a function - and hey, a curly brace indicating a block
+  var varAge = 34; // This is a function scoped variable
 
-For reference, here is the same thing created using the constructor pattern:
+  if (age > 0) { // This is yet another curly brace, and thus a block
+    const constAge = age * 2; // This is a block scoped variable that exists in it's closest outer block, the if's in this case
+    console.log(constAge);
+  }
 
-~~~javascript
-const Person = function(name, age) {
-  this.sayHello = () => console.log('hello!');
-  this.name = name;
-  this.age = age;
-};
+  console.log(constAge); // ERROR! We tried to access a block scoped variable outside its scope
+}
 
-const jeff = new Person('jeff', 27);
+printAge(globalAge);
+console.log(varAge); // ERROR! We tried to access a function scoped outside the function it's defined in
 ~~~
 
-Note that factory functions do not utilize [the prototype](https://www.theodinproject.com/lessons/node-path-javascript-objects-and-object-constructors#the-prototype), which does come with a performance penalty. However, as a general rule, this penalty isn't significant unless you're creating thousands of objects.
+Take a while to brew on that example. In the end, it's not some mind-blowing concept but there's a whole bunch of terms in there - it'll all help us understand the next mammoth - closures.
 
-#### Object shorthand
+### Closures aren't scary
 
-A quick note about line 3 from the factory function example. In 2015, a handy new shorthand for creating objects was added into JavaScript. Without the shorthand, line 3 would have looked something like this:
+The best way to approach this would be to start with an example - take a look at this piece of code below.
 
 ~~~javascript
-return {name: name, age: age, sayHello: sayHello};
+function makeAdding (first) { // "first" is scoped within the makeAdding function
+  return function resulting (second) { // "second" is scoped within the resulting function
+    return first + second;
+  }
+} // but we've not seen an example of a **function** being returned, thus far - how do we use it?
```

Comment:
Let's use quotes instead since md it doesn't work inside a code block
```suggestion
} // but we've not seen an example of a "function" being returned, thus far - how do we use it?
```